src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemScopeManager.cs

```diff
@@ -3,88 +3,60 @@
 
 using Microsoft.DotNet.DarcLib;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace ProductConstructionService.WorkItems;
 
 public class WorkItemScopeManager
 {
-    private readonly AutoResetEvent _autoResetEvent;
     private readonly IServiceProvider _serviceProvider;
-    private readonly ILogger<WorkItemScopeManager> _logger;
-    private WorkItemProcessorState _state;
-
-    public WorkItemProcessorState State
-    {
-        get => _state;
-        private set
-        {
-            if (_state != value)
-            {
-                _state = value;
-                _logger.LogInformation("WorkItemsProcessor state changing to {newValue}", value);
-            }
-        }
-    }
+    private readonly WorkItemProcessorState _state;
+    private readonly int _pollingRateSeconds;
 
     internal WorkItemScopeManager(
-        bool initializingOnStartup,
         IServiceProvider serviceProvider,
-        ILogger<WorkItemScopeManager> logger)
+        WorkItemProcessorState state,
+        bool initializingOnStartup,
+        int pollingRateSeconds)
     {
-        _autoResetEvent = new AutoResetEvent(!initializingOnStartup);
-        _logger = logger;
         _serviceProvider = serviceProvider;
-        _state = initializingOnStartup ? WorkItemProcessorState.Initializing : WorkItemProcessorState.Working;
-    }
+        _state = state;
+        if (initializingOnStartup)
+        {
+            _state.StartInitializingAsync().GetAwaiter().GetResult();
```

Comment:
This is not great to do in a ctor. Especially when we use DI. Can we go around this?

Reply:
I think we could do this in `InitializationBackgroundService`, but we'd have to add some synchronization between it, and the `WorkItemConsumer`, so we don't start trying to process jobs before we set the correct status. And then for the other scenario, where there's no initialization, which only happens in local development, we could add a different service, that would just set this to running

Reply:
Yeah, this should be an explicit call because it could also happen something will request this in the DI before we actually want to do it ourselves. Maybe we could check if there is an ASP.NET core thing for this like a startup filter or similar

Reply:
In the end I went with an easy option of just initializing the state before we call `app.Run()`